gh-126719: Clarify `math.fmod` documentation (#127741)
python/cpython: Doc/library/math.rst

.. function:: fmod(x, y)

>    Return the remainder of division ``x / y``, as defined by the platform C library

Reviewer: ```suggestion
   Return the floating-point remainder of ``x / y``, as defined by the platform C library
```

   function ``fmod(x, y)``. Note that the Python expression ``x % y`` may not return
   the same result.  The intent of the C standard is that ``fmod(x, y)`` be exactly
   (mathematically; to infinite precision) equal to ``x - n*y`` for some integer *n*
   such that the result has the same sign as *x* and magnitude less than ``abs(y)``.
   Python's ``x % y`` returns a result with the sign of *y* instead, and may not be
   exactly computable for float arguments. For example, ``fmod(-1e-100, 1e100)``
   is ``-1e-100``, but the result of Python's ``-1e-100 % 1e100`` is ``1e100-1e-100``,
   which cannot be represented exactly as a float, and rounds to the surprising ``1e100``.

>    For this reason, the function :func:`fmod` is generally preferred when working with

Reviewer: ```suggestion
   For this reason, :func:`fmod` is generally preferred when working with
```

.. function:: fmod(x, y)

>    Return the floating-point remainder of ``x / y``, as defined by the platform C library function ``fmod(x, y)``. Note that the

Reviewer: ```suggestion
   Return the floating-point remainder of ``x / y``,
   as defined by the platform C library function ``fmod(x, y)``. Note that the
```

Is the diff much larger if you do something like this? because now the line is effectively > 80-chars (it's roughly 130 chars I think)

Author: @picnixz I would approve that but Ill see what @skirpichev has to say so that I don't waste my time. In my opinion some formatting is defiantly necessary.
